### PR TITLE
Rename `_device_type` to `_accelerator_type`

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Part of #9932 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @justusschock @awaelchli @akihironitta @rohitgr7

### CODE REVIEW

**Comment by four4fish** [comment]
We have a issue open to remove trainer._device_type, can we rename after finish the deprecation? https://github.com/PyTorchLightning/pytorch-lightning/issues/11002

**Review by awaelchli — APPROVED** [review]

**Comment by awaelchli** [comment]
@four4fish A deprecation is not needed, as the attribute is not public. #11002 and this PR do not block each other as best as I can tell.

File: pytorch_lightning/callbacks/gpu_stats_monitor.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def setup(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", stage: O
         if not trainer.logger:
             raise MisconfigurationException("Cannot use GPUStatsMonitor callback with Trainer that has no logger.")
 
-        if trainer._device_type != _AcceleratorType.GPU:
+        if trainer._accelerator_type != _AcceleratorType.GPU:
```
Comment (ananthsub) [inline pytorch_lightning/callbacks/gpu_stats_monitor.py:129]: (for a separate issue/PR) i know this is a rename, but why do we need this check at all? couldn't we use the strategy's root_device instead?

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/callbacks/gpu_stats_monitor.py (hunk 1 above)
Comment (kaushikb11) [reply pytorch_lightning/callbacks/gpu_stats_monitor.py:129]: Or strategy's `on_gpu` property. Could be a follow up.

**Comment by kaushikb11** [comment]
Need some refactoring for `accelerator_connector`. Converting to Draft.

**Pull request closed without merging**